openwisp_notifications/tasks.py

```diff
@@ -202,3 +211,79 @@ def delete_ignore_object_notification(instance_id):
     Deletes IgnoreObjectNotification object post it's expiration.
     """
     IgnoreObjectNotification.objects.filter(id=instance_id).delete()
+
+
+@shared_task(base=OpenwispCeleryTask)
+def send_batched_email_notifications(instance_id):
+    """
+    Sends a summary of notifications to the specified email address.
+    """
+    if not instance_id:
+        return
+
+    cache_key = f'email_batch_{instance_id}'
+    cache_data = cache.get(cache_key, {'pks': []})
+
+    if not cache_data['pks']:
+        return
+
+    display_limit = app_settings.EMAIL_BATCH_DISPLAY_LIMIT
+    unsent_notifications = Notification.objects.filter(
+        id__in=cache_data['pks']
+    ).order_by('-timestamp')
+    notifications_count = unsent_notifications.count()
+    current_site = Site.objects.get_current()
+    email_id = cache_data.get('email_id')
+
+    # Send individual email if there is only one notification
+    if notifications_count == 1:
+        notification = unsent_notifications.first()
+        send_notification_email(notification)
+    else:
+        # Show the amount of notifications according to configured display limit
+        show_notification_description = notifications_count <= display_limit
```

Comment:
I think we could display the description anyway, but we have to limit the amount of notifications shown

Reply:
Updated!